crates/turborepo-ui/src/tui/preferences.rs

```diff
@@ -0,0 +1,132 @@
+use std::{
+    fs::{self, File},
+    io::{BufReader, Write},
+};
+
+use serde::{Deserialize, Serialize};
+use serde_json::{from_reader, json, Value};
+use turbopath::AbsoluteSystemPathBuf;
+
+use super::task::TasksByStatus;
+
+#[derive(Serialize, Deserialize, Debug)]
+pub struct Preferences {
+    pub is_task_list_visible: Option<bool>,
+    pub active_task: Option<String>,
+    pub is_pinned_task_selection: Option<bool>,
+}
+
+#[derive(Debug)]
+pub enum PreferenceFields {
+    IsTaskListVisible,
+    ActiveTask,
+    PinnedTaskSelection,
+}
+
+impl Default for Preferences {
+    fn default() -> Self {
+        Self {
+            active_task: None,
+            is_task_list_visible: Some(true),
+            is_pinned_task_selection: Some(false),
+        }
+    }
+}
+
+const TUI_PREFERENCES_PATH_COMPONENTS: &[&str] = &[".turbo", "preferences", "tui.json"];
+
+fn read_json(path: &AbsoluteSystemPathBuf) -> Preferences {
+    File::open(path)
+        .ok()
+        .and_then(|file| from_reader(BufReader::new(file)).ok())
+        .unwrap_or_default()
+}
+
+impl Preferences {
+    pub fn update_preference(
+        repo_root: &AbsoluteSystemPathBuf,
+        field: PreferenceFields,
+        new_value: Value,
+    ) -> Result<(), Box<dyn std::error::Error>> {
+        // TODO: Clean these up, should be taken from constants
+        let preferences_dir = repo_root.join_components(&[".turbo", "preferences"]);
+        let preferences_file = repo_root.join_components(&[".turbo", "preferences", "tui.json"]);
+
+        fs::create_dir_all(preferences_dir.as_std_path())?;
+
+        let mut json: Value = if preferences_file.exists() {
+            let json_string = fs::read_to_string(&preferences_file)?;
+            serde_json::from_str(&json_string)?
+        } else {
+            json!({})
+        };
```

Comment:
`exists()` generally is a smell since it can lead to TOCTOU bugs. We can use [`read_existing_to_string`](https://github.com/vercel/turborepo/blob/9fad9cb81e41f5cb9cf8cccf3ee1f279987d6478/crates/turborepo-lib/src/commands/link.rs#L235) to handle defaults.
```suggestion
        let mut json: Value = preferences_file.read_existing_to_string()?.unwrap_or_else(|| "{}".to_string());
```